shared/modules/doc/README.md

```diff
@@ -1,53 +1,66 @@
+	- [Module creation](#module-creation)
+		- [Interacting \& Registering with the `bus`](#interacting--registering-with-the-bus)
+			- [Start the module](#start-the-module)
+			- [Add a logger to the module](#add-a-logger-to-the-module)
+		- [Get the module `bus`](#get-the-module-bus)
+		- [Stop the module](#stop-the-module)
 
-Speak to @andrewnguyen22 or @Olshansk for more details.
+## Definitions
 
-!!! IMPORTANT !!!
+### **Module common interface**
 
-## Implementation
+A module common interface is an interface that defines the methods that we modelled as common to multiple modules. For example: the ability to start/stop a module is pretty common and for that we defined the `InterruptableModule` interface.
+There are some interfaces that are common to multiple modules and we followed the [Interface segregation principle](https://en.wikipedia.org/wiki/Interface_segregation_principle) and also [Rob Pike's Go Proverb](https://youtu.be/PAAkCSZUG1c?t=317):
 
-In order to maintain code agnostic from the inception of the implementation, protobuf3 is utilized for all structures in this package.
+> The bigger the interface, the weaker the abstraction.
 
-It is important to note, that while Pocket V1 strives to not share objects between modules, the genesis module will inevitably overlap between other modules.
+These interfaces that can be embedded in modules are defined in `shared/modules/module.go`. GoDoc comments will provide you with more information about each interface.
 
-Another architecture worth considering and perhaps is more optimal as the project nears mainnet is allowing each module to create and maintain their own genesis object and config files
+### **Module**
 
-### Code Organization
+A module is an abstraction (go interface) of a self-contained unit of functionality that carries out a specific task/set of tasks with the idea of being reusable, modular, testable and having a clear and concise API. A module might implement 0..N common interfaces. You can find additional details in the [Modules in detail](#modules-in-detail) section below.
 
-```bash
-genesis
-├── docs
-│   ├── CHANGELOG.md    # Genesis module changelog
-│   ├── README.md       # Genesis module README
-├── proto
-│   ├── account.proto   # account structure
-│   ├── actor.proto     # actor structure
-│   ├── config.proto    # configuration structure
-│   ├── gov.proto       # params structure
-│   ├── state.proto     # genesis state structure
-├── test_artifacts      # the central point of all testing code (WIP)
-│   ├── generator.go    # generate the genesis and config.json for tests and build
-│   ├── gov.go          # default testing parameters
-```
+### **Module mock**
+
+A mock is a fake implementation of a module that is used for testing purposes. It is used to simulate the behaviour of the module and to verify that the module is interacting with other modules correctly. Mocks are generated using `go:generate` directives together with the `mockgen` tool.
```

Comment:
I  think the existing description is great, I took a quick look at how [how the wikipedia article describes mocks](https://en.wikipedia.org/wiki/Mock_object) and thought it could be useful to add a bit more detail / rephrasing given the ambiguity of the labels on the mock object spectrum.

```suggestion
A mock is a stand-in, a fake or simplified implementation of a module that is used for testing purposes. It is used to simulate the behaviour of the module and to verify that the module is interacting with other modules correctly. Mocks are generated using `go:generate` directives together with the `mockgen` tool.
```

Reply:
Wdyt about also making "`mockgen` tool" a link to the `gomock` docs? The README features a "Running mockgen" section above the fold: https://pkg.go.dev/github.com/golang/mock#section-readme